Gems/ROS2/Code/Source/Manipulation/JointsPositionsComponent.cpp

```diff
@@ -94,19 +94,19 @@ namespace ROS2
             return;
         }
 
-        auto commandIter = message.data.cbegin();
-        for (const auto& jointName : m_jointNames)
+        for(int i=0; i<message.data.size(); i++)
```

Comment:
Actually, it looks like the error is in the original logic. commandIter needs to be incremented as well during the traversal of the m_jointNames list. (from the original line). You could also do
```
for (auto jointNameIter = m_jointNames.cbegin(), auto commandIter = message.data.cbegin();
       jointNameIter != m_jointNames..cend() && commandIter != message.data..cend();
       ++jointNameIter, ++commandIter)
```
and update the `m_jointNames[i]` with *jointNameIter and `message.data[i]` with `*commandIter`